src/nvim/bufwrite.c

```diff
@@ -715,13 +715,47 @@ static int get_fileinfo(buf_T *buf, char *fname, bool overwriting, bool forceit,
   return OK;
 }
 
+/// @return The backup file name
+char *buf_get_backup_name(char *fname, char **dirp, bool no_prepend_dot)
```

Comment:
this is a nice refactor that could be merged pretty quickly if extracted to another PR